src/commands/createVirtualMachine/ImageListStep.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ComputeManagementModels } from '@azure/arm-compute';
+import { AzureWizardPromptStep } from "vscode-azureextensionui";
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import { IVirtualMachineWizardContext } from './IVirtualMachineWizardContext';
+
+type ImageReferenceWithLabel = ComputeManagementModels.ImageReference & { label: string };
+
+export class ImageListStep extends AzureWizardPromptStep<IVirtualMachineWizardContext> {
+
+    public async prompt(context: IVirtualMachineWizardContext): Promise<void> {
+        const placeHolder: string = localize('selectImage', 'Select an image');
+        context.image = (await ext.ui.showQuickPick(this.getAvailableImages().map((ir) => { return { label: ir.label, data: ir }; }), {
+            placeHolder
+        })).data;
+    }
+
+    public shouldPrompt(context: IVirtualMachineWizardContext): boolean {
+        return !context.image;
+    }
+
+    private getAvailableImages(): ImageReferenceWithLabel[] {
+        return [
+            {
+                label: 'Ubuntu Server 18.04 LTS - Gen1',
+                publisher: 'Canonical',
+                offer: 'UbuntuServer',
+                sku: '18.04-LTS',
+                version: 'latest',
+                exactVersion: '18.04.202007290'
```

Comment:
This "exactVersion" property makes me nervous - could get out of date quickly. Is it necessary?

Reply:
I think we can take this out, yeah.  What actually gets used in the linuxConfiguration is just "version" which uses the latest.